### DISCUSSION TITLE
Recalculate bounding box

### DISCUSSION THREAD

**Discussion by danilokrulj** [google/model-viewer#4778] [Q&A]
Hi, has anyone had the need to recalculate objects bounding box at some point? I show/hide parts of the model which changes the models height, but the dimension hotspot guides stay the same. I would like to recalculate the bounding box and redraw the dimension lines when the changes happen.

I guess I could change the values of the hotspots manually for this case but I was wondering is there a better / faster way to do this?

Example images:
![image_2024_05_17T08_28_55_284Z](https://github.com/google/model-viewer/assets/92090684/5f88e5dc-5d7f-4367-958d-2f117c27e4a6)
![image_2024_05_17T08_29_04_729Z](https://github.com/google/model-viewer/assets/92090684/925243b6-6710-4f80-8852-1c6fd0c8b0df)

**Comment by elalish** [comment]
First off, very nice use of MV's advanced features! Please feel free to share your site on our show & tell or on twitter and @modelviewer. 

The bounding box gets recalcuated by [`updateFraming()`](https://modelviewer.dev/docs/#entrydocs-stagingandcameras-methods-updateFraming), but I'm not sure you also want to reframe the camera.

**Reply by danilokrulj** [reply]
Thank you elalish.

Would the bounding box recalculation work if I just set the meshes visibility to false (I feel like it would still be included in the bounding box since its just not visible but its still there)?

I made my case work manually by changing the values for the hotspots on button click but I'm asking for future reference :)

**Reply by elalish** [reply]
mesh visibility isn't part of our public API (yet), so I wouldn't bank on it. By that I mean, we can only hide something by making it transparent, which I'm not sure if that should really change the bounds...

**Reply by henrik-dons** [reply]
Hello all,

Instead of showing / hiding parts, I'm working on changing the mesh size using scale functions.

The change of mesh size is working. But I can't get the hotspot size units to update / redraw just yet.

Any insights on how to update the display sizes after scaling the object?

![invalid-scale-size](https://github.com/user-attachments/assets/c4da2cd6-1d5f-4a82-9dbc-927fcfa507ac)

**Reply by danilokrulj** [reply]
Maybe this post can help you.
https://github.com/google/model-viewer/discussions/4988

**Reply by henrik-dons** [reply]
Thank you for your quick reply! That could work when I know the new scale of the model.

But first, how can I get the updated dimensions of the modelViewer?

It seems to return the same dimensions/scale after changing scale of meshes inside the model as shown in above screenshot.

Is there some recalculate dimensions function? Sort of updateFraming for the size?

Looking forward to your reply

**Reply by danilokrulj** [reply]
Yes, you can call this function to recalculate the dimensions.
(by the way, I'm not sure if you need the [sceneSym] part, I just copied all that out of my old code)

const sceneSym = Object.getOwnPropertySymbols(mw).find(x => x.description === "scene");

function UpdateFrameing(){
  modelViewer[sceneSym].updateShadow();
  modelViewer[sceneSym].queueRender();
}

Also, i just looked at the first reply by elalish, there is a link towards the documentation with updateFraming()

**Reply by henrik-dons** [reply]
Thank you for the reply!

Unfortunately I need the original bounding box / size dimensions to update when a mesh is scaled.
modelViewer.updateFraming() still gives me the original model sizes

So I have this table configurator and I only want to change the size of the table top. 
On changing the length of the table top, the size of the whole model should be re-calculated.

Is that something I can do? I can't seem to make this work yet.

Hope this makes sense
